src/session/session_api.c

```diff
@@ -1806,6 +1806,24 @@ __session_timestamp_transaction(WT_SESSION *wt_session, const char *config)
     API_END_RET(session, ret);
 }
 
+/*
+ * __session_timestamp_transaction_commit --
+ *     WT_SESSION->timestamp_transaction_commit method.
+ */
+static int
+__session_timestamp_transaction_commit(WT_SESSION *wt_session, uint64_t ts)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+    SESSION_API_CALL_NOCONF(session, timestamp_transaction_commit);
```

Comment:
I don't quite follow what you mean here sorry. Are you talking about preventing a timestamp from being altered after the transaction has been prepared?

Reply:
Yes; we have to (I think), explicitly allow this after prepare, otherwise it will fail -- whatever the more heavy-weight API does is probably right, but I haven't looked hard.

Reply:
Thanks for explaining that. I'll take a more detailed look to see exactly what the difference is and where I misunderstood which call to make.